accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,514 @@
+
+The Sigstore client library will include direct vendor support for requesting an OIDC id token from supported CI/CD systems, similar to the existing client libraries. It will also include all the required functionality to interact with Fulcio and Rekor, as well as verify the [Sigstore root keys](https://github.com/sigstore/root-signing) using a [TUF](https://theupdateframework.github.io/) client.
+
+#### Fetching Sigstore hosts from npmjs.org
+The npm CLI will fetch Sigstore config from npmjs.org that contains the public Fulcio and Rekor hosts by default:
+```
+GET https://registry.npmjs.org/-/npm/v1/sigstore-config
+
+{
+  "sigstoreConfig": {
+    "fulcioUrl": "https://fulcio.sigstore.dev",
+    "rekorUrl": "https://rekor.sigstore.dev"
+  }
+}
+```
+
+This would allow future extensions where maintainers can configure the instance being used, making sure the release attestation gets placed on the same Rekor instance as the build attestation. The release attestation will be generated by the registry when the package has been successfully authorized and published.
+
+#### Publishing
+Maintainers will need to enable build provenance generation with a command line flag: `npm publish --build-provenance` in addition to running it in a supported CI/CD system. The command will error with an error message if this is not run in a supported build system.
+
+Once enabled, the CLI will perform the required steps to generate the provenance attestation and upload this to the public Rekor ledger for transparency and a detached copy that contains proof of inclusion from Rekor will be stored in the npm registry.
+
+##### What goes on the public Rekor ledger?
+Only public npm packages will be signed and published to the public [rekor.sigstore.dev](https://docs.sigstore.dev/rekor/overview) ledger by default.
+
+Privately scoped packages, or packages from private repositories, will not be signed or published to the public ledger. This will be determined by interrogating the npm registry at publish time. Users will be able to override this similar to the `cosign --force` command by passing an argument to `npm publish`, e.g. `--force-build-provenance`.
+
+The entry will be stored and retrieved using the shasum of the package tarball. This will make sure we don't find any signatures for a  deleted package that has been resurrected with the same name and version.
+
+There will be no way of deleting an entry if this was published unintentionally to the ledger as it's immutable by design to prevent tampering. The practical consequence of publishing to the ledger is that information about your package (name, version), repository (commit, branch, owner) and workflow (name, id) are uploaded to the public ledger. A possible workaround would be to rename the leaked repo, make it private and delete the old one.
+
+##### Detailed steps to publish
+Interaction between the npm CLI tool running in CI, the npm registry, and the Sigstore components Fulcio and Rekor:
+
+```mermaid
```

Comment:
Do you want to include the OIDC request in this diagram?

Reply:
Good spot, updated